pkg/flavorgen/flavor_gen.go

```diff
@@ -0,0 +1,292 @@
+	// That would permit usages such as
+	//	-f xx.json -f yy.json
+	// and other combinations.
+	for _, template := range strings.Split(value, ",") {
+		*templates = append(*templates, template)
+	}
+
+	return nil
+}
+
+// processJsonFile is used to process the hostmanifest and flavor templates
+// Returns error if could not load the files
+// Returns error if not valid json
+// Returns error if could not unmarshall the json
+// Returns error if all flavor template condition not matches
+func processJsonFile(manifestFilepath string, flavorTemplates []string) (hcType.HostManifest, []hvs.FlavorTemplate, error) {
+	defaultLog.Trace("flavorgen/flavor_gen:processJsonFile() Entering")
+	defer defaultLog.Trace("flavorgen/flavor_gen:processJsonFile() Leaving")
+
+	var hostManifest hcType.HostManifest
+	var flavors []hvs.FlavorTemplate
+
+	// Read the hostmanifestfile
+	hostManifestJSON, err := ioutil.ReadFile(manifestFilepath)
+	if err != nil {
+		return hcType.HostManifest{}, nil, errors.Wrap(err, "flavorgen/flavor_gen:processJsonFile() Could not load host manifest file")
+	}
+
+	if len(hostManifestJSON) == 0 {
+		return hcType.HostManifest{}, nil, errors.Wrap(err, "flavorgen/flavor_gen:processJsonFile() Empty hostmanifest file given, unable to proceed further")
+	}
+
+	// Validate the format
+	if !json.Valid(hostManifestJSON) {
+		return hcType.HostManifest{}, nil, errors.New("flavorgen/flavor_gen:processJsonFile() Hostmanifest file is not a valid json")
+	}
+	err = json.Unmarshal(hostManifestJSON, &hostManifest)
+	if err != nil {
+		fmt.Errorf("Could not unmarshal host manifest json %s", err)
+		return hcType.HostManifest{}, nil, errors.New("flavorgen/flavor_gen:processJsonFile() Could not unmarshal host manifest json")
+	}
+
+	manifest, err := jsonquery.Parse(bytes.NewReader(hostManifestJSON))
+	if err != nil {
+		fmt.Errorf("Could not parse host manifest json %s", err)
+		return hcType.HostManifest{}, nil, errors.Wrap(err, "flavorgen/flavor_gen:processJsonFile() Could not parse host manifest json")
+	}
+
+	for _, template := range flavorTemplates {
+		var flavorTemplate hvs.FlavorTemplate
+		flavorJSON, err := ioutil.ReadFile(template)
+		if err != nil {
+			return hcType.HostManifest{}, nil, errors.Wrap(err, "flavorgen/flavor_gen:processJsonFile() Could not load flavor template file")
+		}
+
+		if len(flavorJSON) == 0 {
+			return hcType.HostManifest{}, nil, errors.Wrap(err, "flavorgen/flavor_gen:processJsonFile() Empty flavor template file is given, unable to proceed further")
+		}
+
+		// Validate the format
+		if !json.Valid(flavorJSON) {
+			return hcType.HostManifest{}, nil, errors.New("flavorgen/flavor_gen:processJsonFile() Given flavor template file is not a valid json")
+		}
+
+		err = json.Unmarshal(flavorJSON, &flavorTemplate)
+		if err != nil {
+			return hcType.HostManifest{}, nil, errors.Wrap(err, "flavorgen/flavor_gen:processJsonFile() Could not unmarshal flavor template json")
+		}

```

Comment:
this can be done in a separate function as this code is repetitive for "hostmanifest" and "template"